docs/examples.rst

```diff
@@ -0,0 +1,310 @@
+Examples
+========
```

Comment:
I think it would be useful to have some high level quick intro outlining what these example cover.

Reply:
Something simple like:
```
The following examples will walk you through the core concepts of YogaDL:
storing, fetching, and streaming datasets.
```
?

Reply:
Yep just this would be good.